ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/rollup/RollupCreatePayload.java

```
@@ -0,0 +1,197 @@
+    this.mergeCoordinator = mergeCoordinator;
+    this.blockResultFactory = blockResultFactory;
+  }
+
+  @Override
+  public String getName() {
+    return RpcMethod.ROLLUP_CREATE_PAYLOAD.getMethodName();
+  }
+
+  @Override
+  public JsonRpcResponse syncResponse(final JsonRpcRequestContext requestContext) {
+    final Object requestId = requestContext.getRequest().getId();
+    final Hash parentBlockHash;
+    final List<Transaction> transactions;
+    final Address suggestedRecipient;
+    final long timestamp;
+    final Bytes32 prevRandao;
+    final List<?> rawTransactions;
+
+    try {
+      parentBlockHash = requestContext.getRequiredParameter(0, Hash.class);
+      rawTransactions = requestContext.getRequiredParameter(1, List.class);
+      prevRandao = requestContext.getRequiredParameter(2, Hash.class);
+      suggestedRecipient = requestContext.getRequiredParameter(3, Address.class);
+      timestamp = Long.decode(requestContext.getRequiredParameter(4, String.class));
+      transactions =
+          rawTransactions.stream()
+              .map(Object::toString)
+              .map(Bytes::fromHexString)
+              .map(TransactionDecoder::decodeOpaqueBytes)
+              .collect(Collectors.toList());
+    } catch (final RLPException | IllegalArgumentException e) {
+      return new JsonRpcErrorResponse(
+          requestContext.getRequest().getId(), JsonRpcError.INVALID_PARAMS);
+    }
+    final Optional<BlockHeader> parentBlock =
+        protocolContext.getBlockchain().getBlockHeader(parentBlockHash);
+    if (parentBlock.isEmpty()) {
+      return replyWithStatus(requestId, RollupCreatePayloadStatus.INVALID_TERMINAL_BLOCK);
+    }
+
+    try {
+      final PayloadCreationResult result =
+          mergeCoordinator.createBlock(
+              parentBlock.get(), timestamp, suggestedRecipient, transactions, prevRandao);
+
+      if (result.getBlockValidationResult().errorMessage.isPresent()) {
+        return new JsonRpcSuccessResponse(
```

Should be an error response? Hiding an error code in a successful response seems bad.

---

By looking into [EngineForkchoiceUpdated](https://github.com/hyperledger/besu/blob/main/ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/EngineForkchoiceUpdated.java#L87) and [EngineNewPayload](https://github.com/hyperledger/besu/blob/main/ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/EngineNewPayload.java#L209) it seems to have a mix of both approaches.

In this case, I was following EngineNewPayload when block validation fails.

If we change to a JSON-RPC error, what do you suggest the response to be like? create a new enum entry `INVALID_PAYLOAD(-38004, "Invalid payload attributes")`

```json
{"jsonrpc": "2.0", "error": {"code": -38004, "message": "Invalid payload", "data": "detail validation error"}, "id": null}
```